app/src/main/java/com/ncnf/database/builder/UserBuilder.java

```diff
@@ -23,31 +26,25 @@
 import static com.ncnf.utilities.StringCodes.USERNAME_KEY;
 import static com.ncnf.utilities.StringCodes.UUID_KEY;
 
-
 public class UserBuilder extends DatabaseObjectBuilder<User>{
 
     @Inject
     DatabaseService db;
 
     @Override
     public User toObject(String uuid, Map<String, Object> data) {
+        String username = (String) data.getOrDefault(USERNAME_KEY, "");
```

Comment:
Does getOrDefault also work on a null Map reference? If it's not the case, shouldn't there be a check that data is not null?
And I guess if null.getOrDefault works fine, you can remove the Objects.requireNonNull in EventBuilder.toObject

Reply:
No you are right, I'll add that.